flycheck.el

```diff
@@ -6239,7 +6239,10 @@ See Info Node `(elisp)Byte Compilation'."
         ;; back-substutition work
         (setq default-directory process-default-directory)
         (with-demoted-errors "Error in checkdoc: %S"
-          (checkdoc-current-buffer t)
+          ;; Checkdoc needs the right syntax table
+          ;; See https://github.com/flycheck/flycheck/issues/833
+          (with-syntax-table emacs-lisp-mode-syntax-table
```

Comment:
Indeed, it would be better to add it. I updated the PR.
The file providing that table changed between 24 and 25 (lisp-mode to elisp-mode). The new code loads elisp-mode and falls back to list-mode, but we could also use a version-based check.